### ISSUE TITLE
[Calendar] NVDA automatically changes its mode to Browse inside Calendar

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3088]
## Current behavior 

NVDA changes its mode from Focus to Browse after leaving first focusable button inside calendar.
In Browse mode `aria-describedby` is not announced, so user would only get label and role announced on elements with indicators and without indicators. 
Keyboard navigation is also harder, as user cannot use all four arrows to navigate in four direction rather they can use only up and down arrow to go to previous and next day.

https://github.com/user-attachments/assets/9be17a30-eb37-4a18-8cde-261118332a7c

Issue is caused by nested interactive element inside grid which is known NVDA issue: https://github.com/nvaccess/nvda/issues/13392 

## Sugested solution

Discussion about how focus can be managed inside grid should take place because this might prevent NVDA from automatically changing mode to Browse.

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
